aws/cloudformation/drone-stack.yml

```diff
@@ -0,0 +1,367 @@
+          Image: drone/agent:1.0.0-rc.5
+          MemoryReservation: 512
+          LogConfiguration:
+            LogDriver: awslogs
+            Options:
+              awslogs-group: !Ref DroneWorkerCloudwatchLogsGroup
+              awslogs-region: !Ref AWS::Region
+              awslogs-stream-prefix: ecs
+          MountPoints:
+            -
+              ContainerPath: /var/run/docker.sock
+              SourceVolume: docker-socket
+          Environment:
+            -
+              Name: DRONE_RPC_SERVER 
+              Value: 
+                !Sub 
+                  - "http://${ElbDnsName}"
+                  - {ElbDnsName: !GetAtt DroneMasterELB.DNSName}
+            -
+              Name: DRONE_RUNNER_CAPACITY
+              Value: 1
+            # ECS secrets manager integration not supported by Cloudformation yet: https://github.com/aws/containers-roadmap/issues/97
+            # Instead, we must use "dynamic references": https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/dynamic-references.html
+            -
+              Name: DRONE_RPC_SECRET
+              Value: '{{resolve:secretsmanager:drone/DRONE_RPC_SECRET}}'
+
+  DroneWorkerEcsService:
+    Type: AWS::ECS::Service
+    DependsOn: LoadBalancerListener
+    Properties:
+      Cluster: !Ref DroneWorkerECSCluster
+      LaunchType: EC2
+      DesiredCount: !Ref NumWorkers
+      TaskDefinition: !Ref DroneWorkerTaskDefinition
+      DeploymentConfiguration:
+        MaximumPercent: 100
+        MinimumHealthyPercent: 0
+      PlacementConstraints:
+        - 
+          Type: distinctInstance
+
+  DroneWorkerBusinessHoursStartScheduledAction:
+    Type: AWS::AutoScaling::ScheduledAction
+    Properties:
+      AutoScalingGroupName: !Ref DroneWorkerEcsInstanceAsg
+      MaxSize: 10
+      MinSize: 10
+      # 13:00 UTC every day of week Monday-Friday
+      Recurrence: 0 13 * * MON-FRI
+
+  DroneWorkerBusinessHoursEndScheduledAction:
+    Type: AWS::AutoScaling::ScheduledAction
+    Properties:
+      AutoScalingGroupName: !Ref DroneWorkerEcsInstanceAsg
+      MaxSize: 1
+      MinSize: 1
+      # 4:00 UTC every day
+      Recurrence: 0 4 * * *
+
+  DroneWorkerScalableTarget:
+    Type: AWS::ApplicationAutoScaling::ScalableTarget
+    Properties:
+      MaxCapacity: 10
+      MinCapacity: 1
+      ResourceId:
+        !Sub
+          - "service/${DroneWorkerECSCluster}/${ServiceName}"
+          - {ServiceName: !GetAtt DroneWorkerEcsService.Name}
+      RoleARN: !Sub "arn:aws:iam::${AWS::AccountId}:role/aws-service-role/autoscaling.amazonaws.com/AWSServiceRoleForAutoScaling"
+      ScalableDimension: ecs:service:DesiredCount
+      ScheduledActions: 
+        -
+          ScalableTargetAction:
+            MinCapacity: 10
+            MaxCapacity: 10
+          # 13:00 UTC every day of week Monday-Friday - see https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/ScheduledEvents.html#CronExpressions
+          Schedule: cron(0 13 ? * MON-FRI *)
```

Comment:
Based on previous discussions, I prefer to keep the templates in pure YAML instead of rendering them with Ruby unless necessary. I dont think it's particularly important to keep the windows up to date for daylights savings, so I'm going to just leave this as-is.